TCAT/Controllers/OptionsViewController.swift

```diff
@@ -242,35 +294,74 @@ class OptionsViewController: UIViewController, UITableViewDelegate, UITableViewD
         }
     }
     
+    //Leave now = all buses that leave at the user's "now" time
     func getValidRoutes(routes: [Route]) -> [Route]{
-//        var validroutes: [Route] = []
-//        for route in routes{
-//            let directions = route.directions
-//            for i in 0..<directions.count{
-//                if let walkDir = directions[i] as? WalkDirection{
-//                    if i == 0{
-//                        if let walkDir = route.directions[0] as? WalkDirection{
-//                            walkDir.calculateWalkingDirections({ (distance, walkTimeInterval) in
-//                                if self.searchTimeType == .leaveat{
-//                                    
-//                                }else{
-//                                    let arriveToStopDate = self.searchTime?.addingTimeInterval(walkTimeInterval)
-//                                    if(arriveToStopDate! <= route.directions[1].time){
-//                                        validroutes.append(route)
-//                                    }
-//                                }
-//                            })
-//                        }
-//                    }else if i == (directions.count - 1){
-//                            
-//                    }else{
-//                            
-//                    }
-//                }
-//            }
-//        }
-        
-        return []
+        var validroutes: [Route] = []
+        for route in routes{
+            var validRoute = true
+            let directions = route.directions
+            //Check directions to invalidate route
+            for i in 0..<directions.count{
+                if let walkDir = directions[i] as? WalkDirection{
+                    if i == 0{
+                        walkDir.calculateWalkingDirections({ (distance, walkTimeInterval) in
+                            //this might be sketch for leave now, check logic
+                            if self.searchTimeType == .leaveat{ //make sure if walk now to stop, get there before leaveat time
```

Comment:
I mentioned in Austin's PR, but ideally these enums could be camel-cased (.leaveAt)